tests/main/cgroup-tracking/task.yaml

```diff
@@ -165,9 +165,11 @@ execute: |
 
     echo "Because the script above used \"sleep 1m\" instead of \"exec sleep 1m\" there"
     echo "are now two processes: the shell and sleep."
-    test "$(wc -l < "${base_cg_path}${pid3_tracking_cg_path}/cgroup.procs")" -eq 2
+    cgroup_procs_path="${base_cg_path}${pid3_tracking_cg_path}/cgroup.procs"
+    test "$(wc -l < "$cgroup_procs_path")" -eq 2
     kill "$pid3_sh"
-    test "$(wc -l < "${base_cg_path}${pid3_tracking_cg_path}/cgroup.procs")" -eq 1
-    wait "$session3_pid" || true  # same as above
+    #shellcheck disable=SC2016
+    cgroup_procs_path="$cgroup_procs_path" retry -n 10 --wait 0.5 sh -c 'test "$(wc -l < $cgroup_procs_path)" -eq 1'
 
+    wait "$session3_pid" || true  # same as above
```

Comment:
Thinking about the original test a bit more, I think the way it's written makes it a bit fragile. We kill the sleep process, for a brief second we observe that there's only the shell process remaining in the cgroup and then wait for it to finish.

As a temporary fix the change is probably ok, but we should rethink the test to make sure that we check snapd things rather than exercise cgroups.